README.md

```diff
@@ -7,3 +7,88 @@ command used for creating typescript template `create-react-app my-app --templat
 
 - run `yarn install` to install all required project dependencies.
 - run `yarn start` to run project.
+
+## Coding guidelines
+
+Developers are always welcome to contribute to this guidelines with clear examples.
+
+---
+
+- Use typescript types as much as possible
+
+  - Avoid using “any” type
+  - Create interfaces for objects (this helps in standardising the structure of complex objects, and promotes reusability) - [Typescript Interfaces](https://www.typescriptlang.org/docs/handbook/interfaces.html)
+
+- Utils file
+
+  - If a utility method is specific only to the component, has logic specific to the component and cannot be used anywhere else, add it to `<component_name>.utils.ts` in the same folder as the component.
+  - Otherwise, add it to one of the files inside `src/common/utils.ts` folder.
+
+- Interfaces file
+
+  - If an interface is specific only to the component, and cannot be used anywhere else
+    - If an interface is small and a component needs 1-2 such interfaces, then the interface can be defined inside the component
+    - Otherwise, use `<component_name>.interfaces.ts` file in the same folder as the component
+  - If an interface can be used by multiple components, methods etc (eg: module specific interface, api related interfaces), define them inside `src/common/interface.ts` folder
+
+- Always use hooks. Do not create any component in classes
+
+  - Older components can remain class based, but no new component should be in class
+  - Never break the rules of hooks - https://reactjs.org/docs/hooks-rules.html.
```

Comment:
I think this is redundant because we assume that the contributor has some React experience.
Moreover any react codebase will need to adhere to these rules so it's a bit implicit.

Reply:
This hasn't been addressed